src/test/java/seedu/address/logic/commands/FilterCommandTest.java

```diff
@@ -0,0 +1,108 @@
+import static seedu.address.logic.commands.CommandTestUtil.assertCommandSuccess;
+import static seedu.address.testutil.TypicalPersons.ALICE;
+import static seedu.address.testutil.TypicalPersons.BENSON;
+import static seedu.address.testutil.TypicalPersons.CARL;
+import static seedu.address.testutil.TypicalPersons.getTypicalAddressBook;
+
+import java.util.Arrays;
+import java.util.List;
+
+import org.junit.jupiter.api.Test;
+
+import seedu.address.model.Model;
+import seedu.address.model.ModelManager;
+import seedu.address.model.UserPrefs;
+import seedu.address.model.person.PersonHasFeaturePredicate;
+import seedu.address.model.person.Phone;
+import seedu.address.model.tag.Tag;
+
+public class FilterCommandTest {
+    private Model model = new ModelManager(getTypicalAddressBook(), new UserPrefs());
+    private Model expectedModel = new ModelManager(getTypicalAddressBook(), new UserPrefs());
+    private PersonHasFeaturePredicate highTagOnlyPredicate =
+          new PersonHasFeaturePredicate(new Tag(VALID_TAG_HIGH_RISK), null);
+    private PersonHasFeaturePredicate lowTagOnlyPredicate =
+          new PersonHasFeaturePredicate(new Tag(VALID_TAG_LOW_RISK), null);
+
+    private PersonHasFeaturePredicate mediumTagOnlyPredicate =
+          new PersonHasFeaturePredicate(new Tag(VALID_TAG_MEDIUM_RISK), null);
+
+    private PersonHasFeaturePredicate phoneOnlyPredicate =
+          new PersonHasFeaturePredicate(null, new Phone(ALICE.getPhone().value));
+
+    private PersonHasFeaturePredicate phoneAndTagPredicate =
+          new PersonHasFeaturePredicate(new Tag(VALID_TAG_HIGH_RISK), new Phone(ALICE.getPhone().value));
+
+    @Test
+    public void equals() {
+
+        FilterCommand findFirstCommand = new FilterCommand(phoneAndTagPredicate);
+        FilterCommand findSecondCommand = new FilterCommand(lowTagOnlyPredicate);
+
+        // same object -> returns true
+        assertTrue(findFirstCommand.equals(findFirstCommand));
+
+        // same values -> returns true
+        FilterCommand findFirstCommandCopy = new FilterCommand(phoneAndTagPredicate);
+        assertTrue(findFirstCommand.equals(findFirstCommandCopy));
+
+        // different types -> returns false
+        assertFalse(findFirstCommand.equals(1));
+
+        // null -> returns false
+        assertFalse(findFirstCommand.equals(null));
+
+        // different command -> returns false
+        assertFalse(findFirstCommand.equals(findSecondCommand));
+    }
+    @Test
+    public void execute_multiplePersonsFound() {
+        String expectedMessage = String.format(MESSAGE_PERSONS_LISTED_OVERVIEW, 2);
+        FilterCommand command = new FilterCommand(highTagOnlyPredicate);
+        expectedModel.updateFilteredPersonList(highTagOnlyPredicate);
+        assertCommandSuccess(command, model, expectedMessage, expectedModel);
+        assertEquals(Arrays.asList(ALICE, BENSON), model.getFilteredPersonList());
```

Comment:
Perhaps instead of using Arrays.asList, you could use List.of() as it's more succinct and doesn't allow null elements, reducing potential issues.